### DISCUSSION TITLE
MQTT problems in HA

### DISCUSSION THREAD

**Discussion by c-mholm** [syssi/homeassistant-goecharger-mqtt#182] [General]
Hi

I have problems with the go-e charger mqtt integration in HA? I can see the topic and messages updating in mqtt Explorer but can’t get the entities to show anything else than unavailable. Integration found the charger, topic is same as in the app. I have tried “/go-eCharger” “go-eCharger/“ “go-eCharger” and “/go-eCharger/“ as topics in integration but get nothing. In the app I have “go-eCharger/“ as topic. It would not connect to mosquitto broker without the slash. Anyone have any ideas what to try?

Thanks.

**Comment by syssi** [comment]
Could you provide some example traffic? I would like to make sure it's API version 2 traffic. Version 1 isn't supported.

**Reply by c-mholm** [reply]
Thanks for helping out. Hope this screen shot can tell you if it’s API2 traffic or not.

![image](https://github.com/user-attachments/assets/32ac040c-6011-49a2-b06e-283b9ce2eced)

![image](https://github.com/user-attachments/assets/09f2a46b-b343-4f16-888e-a885ffc6a7da)

![image](https://github.com/user-attachments/assets/a0f1d393-dff8-402b-bdf4-f91edb5f51e9)

**Reply by syssi** [reply]
It's APIv2 traffic but the serial number as part of the topic is missing. Please change the prefix to `go-eCharger/222222/`.

**Reply by c-mholm** [reply]
Thanks a lot for your help, now it’s working. I did not understand that I had to put the serial in the topic. Read a lot on this topic the whole day, and many things have changed since all the readmes was done. Maybe update the instructions with fresh screen shots and maybe put [go-eCharger/“your serial here”/] or something like that, to make stupid people like me understand how it’s working?

**Reply by syssi** [reply]
The behavior of the go-e firmware has changed over time multiple times. In the past the serial number was a static part of the topic (and no part of the prefix). I will try to improve the documentation. Feel free to make a contribution!
